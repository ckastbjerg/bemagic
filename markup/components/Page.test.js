/* eslint-env jest */
const Page = require('./Page');
const dummyBlock = require('../../dummyBlock');

test('returns markup for a complex component', () => {
    global.config = require('../../config');
    expect(Page(dummyBlock)).toMatchSnapshot();
});
