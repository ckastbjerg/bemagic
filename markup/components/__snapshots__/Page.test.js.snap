exports[`test returns markup for a complex component 1`] = `
"
        <div
            class=\"bemagic-app__page bemagic-page js-page js-block\"
            data-page=\"block\"
        >
            
        <div class=\"bemagic-page__heading bemagic-page-heading\">
            Derived example
            <div class=\"bemagic-page-heading__explain\">
                sophisticated guess regarding the components composition
            </div>
        </div>
        <div
            class=\"bemagic-page__section js-page-section null-background\"
            data-class=\"section-block-full\"
        >
            <div>Example markup derived by bemagic</div>
        </div>
    
            
    
        <div class=\"bemagic-page__heading bemagic-page-heading\">
            block
            <div class=\"bemagic-page-heading__explain\">
                Styled for themes:
            </div>
            
        <div
            class=\"bemagic-theme-button bemagic-theme-button--small theme-inverse background js-page-theme\"
            data-section-ref=\"block\"
            data-theme=\"inverse\"
        >
        </div>
    
        </div>
    
    
        <div class=\"bemagic-page__section js-page-section background\" data-class=\"block\">
            <div class=\"block\">…</div>
            
            
                    <div    class=\"block pseudo-hover\">
                        hover
                    </div>
                
            
                    <div    class=\"block pseudo-hover is-dimmed\">
                        dimmed:hover
                    </div>
                
        </div>
    

            
            <div class=\"bemagic-page__title\">chainable variations</div>
            
    
        <div class=\"bemagic-page__heading bemagic-page-heading\">
            modifier
            <div class=\"bemagic-page-heading__explain\">
                Styled for themes:
            </div>
            
        <div
            class=\"bemagic-theme-button bemagic-theme-button--small theme-inverse background js-page-theme\"
            data-section-ref=\"modifier\"
            data-theme=\"inverse\"
        >
        </div>
    
        </div>
    
    
        <div class=\"bemagic-page__section js-page-section background\" data-class=\"modifier\">
            <div class=\"block block--modifier\">…</div>
            
            
                    <div    class=\"block block--modifier pseudo-hover\">
                        hover
                    </div>
                
            
                    <div    class=\"block block--modifier pseudo-hover is-dimmed\">
                        dimmed:hover
                    </div>
                
        </div>
    

        
            
                    
    
        <div class=\"bemagic-page__heading bemagic-page-heading\">
            element
            <div class=\"bemagic-page-heading__explain\">
                Styled for themes:
            </div>
            
        <div
            class=\"bemagic-theme-button bemagic-theme-button--small theme-inverse background js-page-theme\"
            data-section-ref=\"element\"
            data-theme=\"inverse\"
        >
        </div>
    
        </div>
    
    
        <div class=\"bemagic-page__section js-page-section background\" data-class=\"element\">
            <div class=\"block__element\">…</div>
            
            
                    <div    class=\"block__element pseudo-hover\">
                        hover
                    </div>
                
            
                    <div    class=\"block__element pseudo-hover is-dimmed\">
                        dimmed:hover
                    </div>
                
        </div>
    

                    
    
        <div class=\"bemagic-page__heading bemagic-page-heading\">
            element-modifier
            <div class=\"bemagic-page-heading__explain\">
                Styled for themes:
            </div>
            
        <div
            class=\"bemagic-theme-button bemagic-theme-button--small theme-inverse background js-page-theme\"
            data-section-ref=\"element-modifier\"
            data-theme=\"inverse\"
        >
        </div>
    
        </div>
    
    
        <div class=\"bemagic-page__section js-page-section background\" data-class=\"element-modifier\">
            <div class=\"block__element block__element--element-modifier\">…</div>
            
            
                    <div    class=\"block__element block__element--element-modifier pseudo-hover\">
                        hover
                    </div>
                
            
                    <div    class=\"block__element block__element--element-modifier pseudo-hover is-dimmed\">
                        dimmed:hover
                    </div>
                
        </div>
    

                
        </div>
    "
`;
